### PR TITLE
Try blocking only dfi token

### PR DESCRIPTION
**Pull request by monstrobishi**
Changed to block only DFI token and took out optional parameter

### CODE REVIEW

File: src/masternodes/mn_rpc.cpp [hunk 1]
```diff
@@ -490,18 +490,12 @@ static CAccounts DecodeRecipientsDefaultInternal(CWallet* const pwallet, UniValu
     UniValue recipients(UniValue::VOBJ);
     bool allowSendingToNotOwnAddresses = false;
```
Comment (ShengguangXiao) [inline src/masternodes/mn_rpc.cpp:491]: This parameter can be removed

**Comment by ShengguangXiao** [comment]
I will test with this PR first.

**Comment by monstrobishi** [comment]
test failing will revise

**Comment by ShengguangXiao** [comment]
Test with `accounttoaccount` result:
Send DFI to self                                 Can
Send DFI to other account                Cannot
Send other token to other account   Can

File: src/masternodes/mn_rpc.cpp [hunk 2]
```diff
@@ -488,20 +488,13 @@ static std::vector<CTxIn> GetAuthInputsSmart(CWallet* const pwallet, int32_t txV
...
 static CAccounts DecodeRecipientsDefaultInternal(CWallet* const pwallet, UniValue const& values) {
     UniValue recipients(UniValue::VOBJ);
-    bool allowSendingToNotOwnAddresses = false;
     for (const auto& key : values.getKeys()) {
-        if (key == "enable_external_dfi_token_tx") {
-            allowSendingToNotOwnAddresses = values[key].getBool();
-        } else {
-            recipients.pushKV(key, values[key]);
-        }
+        recipients.pushKV(key, values[key]);
```
Comment (bvbfan) [inline src/masternodes/mn_rpc.cpp:492]: You can remove it, just rename values to recipients

File: test/functional/feature_mine_cached.py [hunk 3]
```diff
@@ -46,19 +46,9 @@ def run_test(self):
...
-
-        self.nodes[0].sendtokenstoaddress({}, to)
-
-        self.nodes[0].unloadwallet('')
-        self.nodes[0].loadwallet('')
-
-        self.nodes[0].sendtokenstoaddress({}, to)
-
-
 if __name__ == '__main__':
```
Comment (bvbfan) [inline test/functional/feature_mine_cached.py:53]: Why it's removed?

Comment (monstrobishi) [reply test/functional/feature_mine_cached.py:53]: You can fix that later I'm just doing a quick patch for @uzyn the basic token functionality is already handled by many other tests so it's not needed here

**Pull request merged**